modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js

```diff
@@ -162,10 +165,73 @@ export default class GPUGridLayer extends CompositeLayer {
     this.setState({weights, gridSize, gridOrigin, cellSize, boundingBox});
   }
 
-  getPickingInfo({info}) {
-    // TODO: perform picking
-    return info;
+  getHashKeyForIndex(index) {
+    const {gridSize, gridOrigin, cellSize} = this.state;
+    const yIndex = Math.floor(index / gridSize[0]);
+    const xIndex = index - yIndex * gridSize[0];
+    // This will match the index to the hash-key to access aggregation data from CPU aggregation results.
+    const latIdx = Math.floor(
+      (yIndex * cellSize[1] + gridOrigin[1] + 90 + cellSize[1] / 2) / cellSize[1]
+    );
+    const lonIdx = Math.floor(
+      (xIndex * cellSize[0] + gridOrigin[0] + 180 + cellSize[0] / 2) / cellSize[0]
+    );
+    return `${latIdx}-${lonIdx}`;
   }
+
+  getPositionForIndex(index) {
+    const {gridSize, gridOrigin, cellSize} = this.state;
+    const yIndex = Math.floor(index / gridSize[0]);
+    const xIndex = index - yIndex * gridSize[0];
+    const yPos = yIndex * cellSize[1] + gridOrigin[1];
+    const xPos = xIndex * cellSize[0] + gridOrigin[0];
+    return [xPos, yPos];
+  }
+
+  getPickingInfo({info, mode}) {
+    const {index} = info;
+    const isPicked = info.picked && index > -1;
```

Comment:
When may we have `picked: true` and `index: -1`? Add a comment?

Reply:
Checking just `index >= 0` should be enough, same as how `Layer` class does.